stdlib/public/core/StringTesting.swift

```diff
@@ -72,5 +86,93 @@ extension _StringGuts {
     }
     fatalError()
   }
-}
 
+
+/*
+
+ Deconstruct the string into contiguous UTF-8, allocating memory if necessary
+
+┌────────────────────╥───────────────────────┬─────────────────────┬─────────────┬─────────────────┐
+│ Form               ║ owner                 │ pointer+length      │ usesScratch │ allocatedMemory │
+├────────────────────╫───────────────────────┼─────────────────────┼─────────────┼─────────────────┤
+│ small with scratch ║ nil                   │ `scratch`           │ true        │ false           │
+├────────────────────╫───────────────────────┼─────────────────────┼─────────────┼─────────────────┤
+│ small w/o scratch  ║ extra allocation      │ `owner` pointer     │ false       │ true            │
+╞════════════════════╬═══════════════════════╪═════════════════════╪═════════════╪═════════════════╡
+│ immortal, large    ║ nil                   │ literal pointer     │ false       │ false           │
+├────────────────────╫───────────────────────┼─────────────────────┼─────────────┼─────────────────┤
+│ native             ║ __StringStorage       │ tail alloc pointer  │ false       │ false           │
+╞════════════════════╬═══════════════════════╪═════════════════════╪═════════════╪═════════════════╡
+│ shared             ║ __SharedStringStorage │ shared pointer      │ false       │ false           │
+├────────────────────╫───────────────────────┼─────────────────────┼─────────────┼─────────────────┤
+│ shared, bridged    ║ _CocoaString          │ cocoa ASCII pointer │ false       │ false           │
+╞════════════════════╬═══════════════════════╪═════════════════════╪═════════════╪═════════════════╡
+│ foreign            ║ extra allocation      │ `owner` pointer     │ false       │ true            │
+└────────────────────╨───────────────────────┴─────────────────────┴─────────────┴─────────────────┘
+
+*/
+  //@_transparent? constant evaluable?
+  @_alwaysEmitIntoClient
```

Comment:
I think you can remove constant evaluable comment here, if you like. This is not required to be constant evaluable and possibly won't ever be.